File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int>v_s(26);
        vector<int>v_t(26);
        for (char ch:s){
            v_s[int(ch)-97]+=1;
        }
        for (char ch:t){
            v_t[int(ch)-97]+=1;
        }
        for (int i=0;i<26;i++){
            if (v_s[i]!=v_t[i]){return false;}
        }
        return true;
        
    }
};

File: Data Structures & Algorithms/meeting-schedule-ii/submission-2.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

class Solution {
    public int minMeetingRooms(List<Interval> intervals) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        int n = intervals.size();

        Collections.sort(intervals,(a,b)->{
            if (a.start==b.start){
                return Integer.compare(b.end,a.end);
            }
            return Integer.compare(a.start,b.start);
        });
        
        int count=0;

        for (int i=0;i<n;i++){
            int start1 = intervals.get(i).start;
            int end1=intervals.get(i).end;
            if (!pq.isEmpty()){
                if (pq.peek()>start1){
                    // System.out.println(count+" >: "+pq.peek());
                    count++;}
                else{
                    // System.out.println(count+" : "+pq.peek());
                    pq.poll();}
            } else{
                count++;
            }
            pq.add(end1);
        }
        return count;

    }
}
